files/build_templates/instance/swss.service.j2

```diff
@@ -1,12 +1,26 @@
 [Unit]
 Description=switch state service
+{% if multi_instance == 'true' %}
+Requires=database@%i.service
+   {% if sonic_asic_platform == 'vs' %}
+Requires=topology.service
+   {% endif %}
```

Comment:
Why introduce this new dependency? and why single instance does not require?

If it is a general bug fix, please submit another PR.

Reply:
Each multi instance service will have to come up only after database of that particular namespace comes up (database@%i), as database container's network stack is used by other services to come up in the same namespace. So we need this dependency.

Reply:
I mean: Why single instanace on vs platform does not need dependence of topology?

---
In reply to: [391308213](https://github.com/Azure/sonic-buildimage/pull/3888#discussion_r391308213) [](ancestors = 391308213)

Reply:
in mulit-asic-vs platform, topology.sh script is used to create internal topology to mimic asic to asic connectivity, which is not required in single asic vs platform.